providers/oidc.go

```diff
@@ -285,21 +288,36 @@ func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface
 	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
 	if rawGroups != nil && ok {
 		for _, rawGroup := range rawGroups {
-			group, ok := rawGroup.(string)
-			if ok {
-				groups = append(groups, group)
+			formattedGroup, err := formatGroup(rawGroup)
+			if err != nil {
+				logger.Errorf("unable to format group of type %s with error %s", reflect.TypeOf(rawGroup), err)
+				continue
 			}
+			groups = append(groups, formattedGroup)
 		}
 	}
 
 	return groups
+
+}
+
+func formatGroup(rawGroup interface{}) (string, error) {
+	group, ok := rawGroup.(string)
+	if !ok {
+		jsonGroup, err := json.Marshal(rawGroup)
+		if err != nil {
+			return "", err
+		}
+		group = string(jsonGroup)
+	}
+	return group, nil
```

Comment:
This function is a tad terse, it's not obvious what this is trying to achieve, maybe could have done with a comment on this 🤔 

Is the point that `groups` is always flattened to some JSON string?

Reply:
Yeah - only if the IdP had a structure that wasn't `[]string` .  In the contributers case, this was what the claim looked like from there IdP raised in the linked issue:

```
"groups": [
    {
      "groupId": "CIDAAS_ADMINS",
      "roles": [
        "ADMIN"
      ]
    }
  ],
```